mypy_requirements.txt

```diff
@@ -1,15 +1,4 @@
 # TODO(#13131): Merge this file into requirements.txt after python3 migration.
 # These packages are python3 packages to be used by MyPy.
-mypy==0.902
-typed-ast==1.4.2
-types-backports==0.1.3
-types-bleach==3.3.3
-types-certifi==0.1.4
-types-enum34==0.1.8
-types-futures==0.1.6
-types-protobuf==0.1.13
-types-PyYAML==0.1.6
-types-redis==3.5.4
-types-requests==2.25.0
-types-six==0.1.7
+mypy==0.812
 typed-ast==1.4.2
```

Comment:
Why are we leaving `typed-ast` here? It looks like it was duplicated in the original file.

Also, if we do remove `typed-ast`, this fill will only contain `mypy`, which seems unnecessarily complex. Why not just install mypy directly from the code and skip the requirements file?

Reply:
Also, this change means that `mypy`'s dependencies will no longer be pinned, which could allow for automatic upgrades that break our tests. What's the benefit of this change?

Reply:
The reason for this change is that from mypy >= 0.900 requires stubs to be installed as separate pip packages. And right now we were trying to install stub packages in third_party directory and mypy was not able to detect them. So this is the fix we use as installing separate stubs for mypy is a new feature and we expect this feature to be documented well or fixed. Thanks!

Reply:
> Why are we leaving typed-ast here? It looks like it was duplicated in the original file.

typed-ast is required for checks to run on py2 codebase.

Reply:
> Also, this change means that mypy's dependencies will no longer be pinned

Every version of mypy comes with specific stubs. So I don't think this will upgrade the stubs automatically.
Please let me know if I got you wrong. Thanks!